### PR TITLE
Included help message when a required value is missing.

### LINKED ISSUE #314: Include help message when a required value is missing
As with #282, the error output when a user doesn't provide a required value should include the help text for that argument or option.

```swift
struct Example: ParsableCommand {
    @Argument(help: "The number of items")
    var count: Int

    @Option(help: "The verbosity level")
    var verbosity: Int
}
```

**Existing:**

```
$ example --verbosity
Error: Missing value for '--verbosity <verbosity>'
Usage: example <count> --verbosity <verbosity>
  See 'example --help' for more information.
$ example --verbosity 3
Error: Missing expected argument '<count>'
Usage: example <count> --verbosity <verbosity>
  See 'example --help' for more information.
```

**Improved:**

```
$ example --verbosity
Error: Missing value for '--verbosity <verbosity>'
Help:  --verbosity <verbosity>  The verbosity level 
Usage: example <count> --verbosity <verbosity>
  See 'example --help' for more information.
$ example --verbosity 3
Error: Missing expected argument '<count>'
Help:  <count>  The number of items 
Usage: example <count> --verbosity <verbosity>
  See 'example --help' for more information.
```

Note that the existing behavior of printing the full help along with an error message should be maintained if the user provides _no_ arguments at all:

```
$ example
Error: Missing expected argument '<count>'

USAGE: example <count> --verbosity <verbosity>

ARGUMENTS:
  <count>                 The number of items

OPTIONS:
  --verbosity <verbosity> The verbosity level
  -h, --help              Show help information.
```

### PR DESCRIPTION
**Pull request by dirtyhabits97**
Hi, these changes should fix #314.

### Checklist
- [x] I've added at least one test that validates that my change is working, if appropriate
- [x] I've followed the code style of the rest of the project
- [x] I've read the [Contribution Guidelines](CONTRIBUTING.md)
- [x] I've updated the documentation if necessary

### CODE REVIEW

File: Sources/ArgumentParser/Usage/UsageGenerator.swift [hunk 1]
```diff
@@ -369,6 +373,25 @@ extension ErrorMessageGenerator {
...
     }
   }
+
+  func missingValueForOptionHelpMessage(name: Name) -> String {
+    guard let arg = arguments.first(where: { $0.names.contains(name) }) else {
+      return ""
+    }
+
+    let help = arg.help.abstract
+    return "\(name.synopsisString) <\(arg.valueName)> \(help)"
```
Comment (natecook1000) [inline Sources/ArgumentParser/Usage/UsageGenerator.swift:383]: Nitpick — here and below, can you add in one more space between the value name and the abstract?
```suggestion
    return "\(name.synopsisString) <\(arg.valueName)>  \(help)"
```

**Review by natecook1000 — COMMENTED** [review]
Thanks so much, @dirtyhabits97! Just a formatting tweak in the new help message, then this should be ready to merge.

File: Sources/ArgumentParser/Usage/UsageGenerator.swift (hunk 1 above)
Comment (dirtyhabits97) [reply Sources/ArgumentParser/Usage/UsageGenerator.swift]: done ✅

**Comment by rauhul** [comment]
@swift-ci Please test

**Review by natecook1000 — APPROVED** [review]

**Pull request merged**